### PR TITLE
matrix-zulip-bridge: init at 0.3.0

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

https://github.com/GearKite/MatrixZulipBridge

NixOS module arrive just after.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/matrix-zulip-bridge/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'emoji = "^2.9"' 'emoji = "^2"' \
+  '';
+
+  nativeBuildInputs = [
+    python3.pkgs.poetry-core
+  ];
```
Comment (h7x4) [inline pkgs/by-name/ma/matrix-zulip-bridge/package.nix:25]: Would it still work using the relax deps hook instead?

```suggestion
  nativeBuildInputs = [
    python3.pkgs.poetry-core
    python3.pkgs.pythonRelaxDepsHook
  ];
  
  pythonRelaxDeps = [ "emoji" ];
```

File: pkgs/by-name/ma/matrix-zulip-bridge/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    pytest-cov
+    python-dotenv
+    ruamel-yaml
+    zulip
+  ];
+
+  pythonImportsCheck = [ "matrixzulipbridge" ];
+
+  meta = with lib; {
+    description = "";
```
Comment (h7x4) [inline pkgs/by-name/ma/matrix-zulip-bridge/package.nix:43]:
```suggestion
    description = "A Matrix puppeting appservice bridge for Zulip";
```

File: pkgs/by-name/ma/matrix-zulip-bridge/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    homepage = "https://github.com/GearKite/MatrixZulipBridge";
```
Comment (h7x4) [inline pkgs/by-name/ma/matrix-zulip-bridge/package.nix:44]:
```suggestion
    homepage = "https://github.com/GearKite/MatrixZulipBridge";
    changelog = "https://github.com/GearKite/MatrixZulipBridge/releases/tag/${src.rev}";
```

File: pkgs/by-name/ma/matrix-zulip-bridge/package.nix (hunk 1 above)
Comment (RaitoBezarius) [reply pkgs/by-name/ma/matrix-zulip-bridge/package.nix:25]: Probably, I guess.

**Pull request closed without merging**